app/Http/Controllers/Api/v1/EntityTransformApiController.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Http\Controllers\Api\v1;
+
+use App\Models\Campaign;
+use App\Models\Entity;
+use App\Http\Requests\TransformEntity as Request;
+use App\Services\Entity\TransformService;
+
+class EntityTransformApiController extends ApiController
+{
+    protected TransformService $service;
+
+    public function __construct(TransformService $service)
+    {
+        $this->middleware('auth');
+
+        $this->service = $service;
+    }
+
+    /**
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+    public function transform(Request $request, Campaign $campaign)
+    {
+        $this->authorize('access', $campaign);
+
+        foreach ($request->entities as $id) {
```

Comment:
No permission check. I can transform as a user ALL entities of a campaign now, despite having 0 permissions to edit anything :-) Works on a public campaign I'm not a member of too